src/lib.rs

```diff
@@ -246,13 +251,13 @@ where
 
     fn set(&mut self, path: &str, data: &[u8]) -> Result<usize, Error> {
         let mut de = serde_json_core::de::Deserializer::new(data);
-        self.set_path(&mut path.split(Self::SEPARATOR).peekable(), &mut de)?;
+        self.set_path(&mut path.split(Self::SEPARATOR).skip(1), &mut de)?;
```

Comment:
Would be useful to document what the skip is for (preceding prefix). Alternatively, we _could_ make it accept both a preceding prefix and only pop it off if it's present? Seems like a good idea to me.

Reply:
I prefer the simple control flow and unambiguous requirements.
Not skipping and instead stripping breaks e.g. the case of a bare `Option` (see `tests::option::just_option`).